UPGRADING.md

```diff
@@ -1,6 +1,19 @@
 Upgrading Grape
 ===============
 
+### Upgrading to >= 1.4.1
+
+Previous to 1.3.3, the `declared` helper would always return the complete params structure if `include_missing=true` was set. In 1.3.3 the behavior was changed, so a missing Hash with or without nested parameters would always resolve as `{}`.
+
+In 1.4.1 this behavior is reverted, so the whole params structure will always be available via `declared`, regardless of whether any params are passed.
+
+The following rules now apply to the `declared` helper when params are missing and `include_missing=true`:
+
+* Hash params with childern will resolve as a hash with keys for each declared child.
+* Hash params with no children will resolve as `{}`.
+* Array params will resolve as `[]`.
+* All other params will resolve as `null`.
```

Comment:
Add a simple before/after example.

Add a link to the PR, `For more information see [#2103](https://github.com/ruby-grape/grape/pull/2103).`. We've been forgetting those, would be nice to re-add where missing below.